Adds detection function for pxa168 S0 revision, ensuring that we set pin config registers correctly?
Borrowed from Chumby 2.6.28 kernel
diff -rup linux-3.13-rc1/arch/arm/mach-mmp/include/mach/cputype.h linux-3.13-rc1/arch/arm/mach-mmp/include/mach/cputype.h
--- linux-3.13-rc1/arch/arm/mach-mmp/include/mach/cputype.h	2013-11-22 11:30:55.000000000 -0800
+++ linux-3.13-rc1/arch/arm/mach-mmp/include/mach/cputype.h	2013-11-29 14:42:56.829616381 -0800
@@ -2,6 +2,7 @@
 #define __ASM_MACH_CPUTYPE_H
 
 #include <asm/cputype.h>
+#include <mach/addr-map.h>
 
 /*
  *  CPU   Stepping   CPU_ID      CHIP_ID
@@ -27,8 +28,17 @@ static inline int cpu_is_pxa168(void)
 	return (((read_cpuid_id() >> 8) & 0xff) == 0x84) &&
 		((mmp_chip_id & 0xfff) == 0x168);
 }
+static inline int cpu_is_pxa168_S0(void)
+{
+	unsigned int chip_id = __raw_readl((AXI_VIRT_BASE + 0x82c00));
+	if (cpu_is_pxa168() && ((chip_id & 0x0000ffff) == 0x0000c910))
+		return 1;
+	else
+		return 0;
+}
 #else
 #define cpu_is_pxa168()	(0)
+#define cpu_is_pxa168_S0() (0)
 #endif
 
 /* cpu_is_pxa910() is shared on both pxa910 and pxa920 */
